src/test/java/junit/org/rapidpm/frp/model/QuintTest.java

```diff
@@ -0,0 +1,126 @@
+package junit.org.rapidpm.frp.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+import org.rapidpm.frp.model.Quint;
+
+public class QuintTest {
+
+  @Test
+  public void testEquals() {
+    final Quint quint = new Quint(0, 1, 2, 3, 4);
+
+    assertTrue(quint.equals(quint));
+    assertFalse(quint.equals(null));
+    assertFalse(quint.equals(new Quint(null, 1, 2, 3, 4)));
+    assertFalse(quint.equals(new Quint(0, null, 2, 3, 4)));
+    assertFalse(quint.equals(new Quint(0, 1, null, 3, 4)));
+    assertFalse(quint.equals(new Quint(0, 1, 2, null, 4)));
+  }
+
+  @Test
+  public void testEqualsT1() {
+    final Quint quint = new Quint(null, 1, 2, 3, 4);
+
+    assertTrue(quint.equals(new Quint(null, 1, 2, 3, 4)));
+    assertFalse(quint.equals(new Quint(0, 1, 2, 3, 4)));
+  }
+
+  @Test
+  public void testEqualsT2() {
+    final Quint quint = new Quint(0, null, 2, 3, 4);
+
+    assertTrue(quint.equals(new Quint(0, null, 2, 3, 4)));
+    assertFalse(quint.equals(new Quint(0, 1, 2, 3, 4)));
+  }
+
+  @Test
+  public void testEqualsT3() {
+    final Quint quint = new Quint(0, 1, null, 3, 4);
+
+    assertTrue(quint.equals(new Quint(0, 1, null, 3, 4)));
+    assertFalse(quint.equals(new Quint(0, 1, 2, 3, 4)));
+  }
+
+  @Test
+  public void testEqualsT4() {
+    final Quint quint = new Quint(0, 1, 2, null, 4);
+
+    assertTrue(quint.equals(new Quint(0, 1, 2, null, 4)));
+    assertFalse(quint.equals(new Quint(0, 1, 2, 3, 4)));
+  }
+
+  @Test
+  public void testEqualsT5() {
```

Comment:
Same `TNNull` thing as with Quad imo for the test names

Reply:
As above